cpp-client-reference/src/volition_shim.cpp

```diff
@@ -1,68 +1,304 @@
+
+using namespace empowerops::volition;
+using namespace empowerops::volition::dto;
+using namespace google;
+
+#define GUARD(cond)      \
+    if (!(cond)) return; \
+    else static_cast<void>(0) // NOLINT(readability-else-after-return)
+#define GUARD_R(cond)       \
+    if (!(cond)) return {}; \
+    else static_cast<void>(0) // NOLINT(readability-else-after-return)
+
+template<class T>
+concept Pointer = std::is_pointer_v<T>;
+template<class T>
+concept NonPointer = !Pointer<T>;
+
+template<NonPointer T>
+constexpr auto into_unique(Pointer auto p) -> std::unique_ptr<T> {
+    return std::unique_ptr<T>{reinterpret_cast<T*>(p)};
+}
+
+template<NonPointer T, Pointer U>
+constexpr auto allocate_opaque() noexcept -> U {
+    return reinterpret_cast<U>(std::make_unique<T>().release());
+}
+
+auto continuous_set_lower_bound(continuous_t continuous, double lower_bound) noexcept -> void {
+    GUARD(continuous);
+    reinterpret_cast<ContinuousDTO*>(continuous)->set_lower_bound(lower_bound);
+}
+
+auto continuous_set_upper_bound(continuous_t continuous, double upper_bound) noexcept -> void {
+    GUARD(continuous);
+    reinterpret_cast<ContinuousDTO*>(continuous)->set_upper_bound(upper_bound);
+}
+
+auto input_parameter_set_name(input_parameter_t input, char const* name) noexcept -> void {
+    GUARD(input && name);
+    reinterpret_cast<InputParameterDTO*>(input)->set_name(name);
+}
+
+auto input_parameter_set_continuous(input_parameter_t input) noexcept -> continuous_t {
+    GUARD_R(input);
+    auto* msg = reinterpret_cast<InputParameterDTO*>(input);
+    return reinterpret_cast<continuous_t>(msg->mutable_continuous());
+}
+
+auto variable_mapping_set_inputs(
+    variable_mapping_t mapping,
+    char const* simulation_name,
+    char const* optimization_name) noexcept -> void {
+    GUARD(mapping && simulation_name && optimization_name);
+    auto* map = reinterpret_cast<VariableMappingDTO*>(mapping)->mutable_inputs();
+    (*map)[simulation_name].set_value(optimization_name);
+}
+
+auto variable_mapping_set_outputs(
+    variable_mapping_t mapping,
+    char const* simulation_name,
+    char const* optimization_name) noexcept -> void {
+    GUARD(mapping && simulation_name && optimization_name);
+    auto* map = reinterpret_cast<VariableMappingDTO*>(mapping)->mutable_outputs();
+    (*map)[simulation_name].set_value(optimization_name);
+}
+
+auto simulation_input_parameter_set_name(
+    simulation_input_parameter_t input,
+    char const* name) noexcept -> void {
+    GUARD(input && name);
+    reinterpret_cast<SimulationInputParameterDTO*>(input)->set_name(name);
+}
+
+auto simulation_output_parameter_set_name(
+    simulation_output_parameter_t output,
+    char const* name) noexcept -> void {
+    GUARD(output && name);
+    reinterpret_cast<SimulationOutputParameterDTO*>(output)->set_name(name);
+}
+
+auto simulation_output_parameter_set_is_boolean(
+    simulation_output_parameter_t output,
+    bool boolean) noexcept -> void {
```

Comment:
I think I tried changing it to the format you described at one point, but realized that it would format function prototypes that way as well:
```C++
VLAPI_PUBLIC void simulation_output_parameter_set_is_boolean(
    simulation_output_parameter_t output,
    bool boolean
) VLAPI_NOEXCEPT;
```
which is pretty ugly in my opinion and makes it harder to distinguish from a `struct` definition at first glance.